### PR TITLE
Add octavia net templates including routes

### PR DESCRIPTION
**Pull request by beagles**
Allow configuring routes on multus network configuration. Initially used for octavia network attachment.

As a pull request owner and reviewers, I checked that:
- [x] Appropriate testing is done and actually running
- [x] Appropriate documentation exists and/or is up-to-date:
  - [x] README in the role
  - [x] Content of the docs/source is reflecting the changes

### CODE REVIEW

**Comment by beagles** [comment]
Would it make more sense to simply add this change to the common template? It's not a big change and would avoid having to create an env for each situation where octavia is deployed

**Comment by cjeanner** [comment]
@beagles heya! Sooo we have two ways I think:
- add that change to the "common" with proper conditions
- use symlinks so that you have only one file, and symlinks for the other scenarios needing it
- add that change to the "common" with proper conditions and starting to leverage "imports" jinja2 call to make the template more... "readable" (that's something we have on the roadmap)

If it's easy to add the change to the common with enough security/safeguard, that's probably better. Regarding snippets, as said we have something post-GA to rework a bit those monster templates, sooo... up to you if you want to start using it.

**Comment by lewisdenny** [comment]
This patch was created before support was added to net_mapper for Multus routes [1]. Some small changes are required to match the output. Example from the WIP patch [2]

[1] https://github.com/openstack-k8s-operators/ci-framework/pull/1914
[2] https://github.com/openstack-k8s-operators/ci-framework/blob/632185732f1cc005c38ee8b4194c17510bf42773/tests/unit/module_utils/test_files/networking-definition-valid-all-tools-dual-stack-full-map-out.json#L19

**Review by hjensas — COMMENTED** [review]
/lgtm

Does this need https://github.com/openstack-k8s-operators/ci-framework/pull/1914 to merge first?

**Comment by beagles** [comment]
> /lgtm
> 
> Does this need #1914 to merge first?

I think it will be fine (just a no-op) without it but it's probably best to merge 1914 first.

File: roles/ci_gen_kustomize_values/templates/uni01alpha/network-values/values.yaml.j2 [hunk 1]
```diff
@@ -95,6 +95,16 @@ data:
         "ipam": {
           "type": "whereabouts",
           "range": "{{ network.network_v4 }}",
+{%  if network.tools.multus.ipv4_routes is defined and network.tools.multus.ipv4_routes | length > 0 %}
```
[the remark this answers is not in the document]
Comment (lewisdenny) [reply roles/ci_gen_kustomize_values/templates/uni01alpha/network-values/values.yaml.j2:98]: I think one of the other uni* jobs will be testing IPV6, so one of the network-values files should be looping IPV6 right?

[the remark this answers is not in the document]
Comment (hjensas) [reply roles/ci_gen_kustomize_values/templates/uni01alpha/network-values/values.yaml.j2:98]: correct uni01alpha will not be IPv6 so this is ok. We can do a follow up to handle ipv6 / dual stack if we decide to change the jobs in the future.

**Review by hjensas — COMMENTED** [review]
/lgtm

**Comment by lewisdenny** [comment]
https://github.com/openstack-k8s-operators/ci-framework/pull/1914 is now merged, my final question is will merging this effect the other uni05/06/07 jobs? It seems it would looking at the `net-def.yaml` for those jobs. 

I will ping the dev-ops team to review.

**Comment by hjensas** [comment]
> #1914 is now merged, my final question is will merging this effect the other uni05/06/07 jobs? It seems it would looking at the `net-def.yaml` for those jobs.
> 
> I will ping the dev-ops team to review.

This change is touching these files:
- roles/ci_gen_kustomize_values/templates/uni01alpha/network-values/values.yaml.j2
- roles/ci_gen_kustomize_values/templates/uni05epsilon/network-values/values.yaml.j2
- roles/ci_gen_kustomize_values/templates/uni06zeta/network-values/values.yaml.j2
- roles/ci_gen_kustomize_values/templates/uni07eta/network-values/values.yaml.j2

Is'nt the usage of these files based on the `cifmw_architecture_scenario` variable?
This is hos I undestand [this code](https://github.com/openstack-k8s-operators/ci-framework/blob/929011f6576b3874d96be53959c3ba60d5b654ba/roles/ci_gen_kustomize_values/tasks/generate_snippets.yml#L107-L118)
```
- name: Generate CI snippet
  when:
    - _tmpl_check_path | length > 0
  vars:
    _tmpl_name: >-
      {{ cifmw_ci_gen_kustomize_values_src_file | basename }}.j2
    _tmpl_path: >-
      {{
        (role_path, 'templates', cifmw_architecture_scenario, values_datatype, _tmpl_name) | path_join
      }}
    _common_tmpl_path: >-
      {{
        (role_path, 'templates', 'common', values_datatype, _tmpl_name) | path_join
      }}
    _tmpl_check_path: >-
      {{
        [_tmpl_path, _common_tmpl_path] | select('exists')    # <-- uses archicture scenario specific if exist, else the common template.
      }}
  ansible.builtin.template:
    backup: true
    dest: >-
      {{
        (snippet_datadir,
         '02_ci_data.yaml') | path_join
      }}
    src: "{{ _tmpl_check_path | first }}"
    mode: "0644"
```

i.e this change should only affect uni01alpha, uni05epsilon, uni06zeta and uni07eta ?

**Review by hjensas — COMMENTED** [review]
/lgtm

File: roles/ci_gen_kustomize_values/templates/uni05epsilon/network-values/values.yaml.j2 [hunk 2]
```diff
@@ -0,0 +1,141 @@
+---
+# source: uni06zeta/network-values/values.yaml.j2
```
Comment (hjensas) [inline roles/ci_gen_kustomize_values/templates/uni05epsilon/network-values/values.yaml.j2:2]: s/uni06zeta/uni05epsilon/

**Review by hjensas — CHANGES_REQUESTED** [review]

File: roles/ci_gen_kustomize_values/templates/uni05epsilon/network-values/values.yaml.j2 (hunk 2 above)
Comment (beagles) [reply roles/ci_gen_kustomize_values/templates/uni05epsilon/network-values/values.yaml.j2]: updated

**Comment by lewisdenny** [comment]
This patch looks good to me, I would just like a lgtm from someone on the devops team. 

/cc @sdatko @arxcruz

**Comment by beagles** [comment]
Tested that network attachments were configured as required and controlplane deployed by testproject MR 474.

**Review by lewisdenny — APPROVED** [review]
/lgtm

**Pull request merged**